src/ray/gcs/gcs_server/gcs_health_check_manager.h

```diff
@@ -108,8 +108,8 @@ class GcsHealthCheckManager : public std::enable_shared_from_this<GcsHealthCheck
   /// It can be updated to support streaming call for efficiency.
   class HealthCheckContext {
    public:
-    HealthCheckContext(std::shared_ptr<GcsHealthCheckManager> manager,
-                       std::shared_ptr<grpc::Channel> channel,
+    HealthCheckContext(const std::shared_ptr<GcsHealthCheckManager> &manager,
+                       const std::shared_ptr<grpc::Channel> &channel,
                        NodeID node_id)
         : manager_(manager),
```

Comment:
```suggestion
        : manager_(std::move(manager)),
```

This gives you the same perf gain.

Reply:
Same move for `stub_ = grpc::health::v1::Health::NewStub(channel);`

Reply:
> This gives you the same perf gain.

I think this won't give benefits, `_manager`is actually weak pointer that takes const as argument.